src/maliput_osm/osm/conversions.cc

```diff
@@ -109,6 +112,60 @@ std::unordered_map<T, U> IntersectMaps(const std::unordered_map<T, U>& map1, con
   return intersection;
 }
 
+// Get the 2d-tangent of the @p lanelet at the selected @p lane_end_which.
+// As the lanelet is defined by left and right boundaries, the tangent of the lanelet is calculated using arbitrarily
+// the left bound.
+maliput::math::Vector2 Get2DTangentAtLaneEnd(const lanelet::ConstLanelet lanelet,
+                                             const maliput::api::LaneEnd::Which& lane_end_which) {
+  // Using left bound as default.
+  const LineString3d line_string = ToMaliput(lanelet.leftBound());
+  const double p = lane_end_which == maliput::api::LaneEnd::Which::kStart ? 0. : line_string.length();
+  return utility::Get2DTangentAtP(line_string, p);
+}
+
+// Get the delta of the angle between two vectors.
+template <typename T>
+double GetDiffAngle(const T& lhs, const T& rhs) {
+  return std::acos(lhs.dot(rhs) / (lhs.norm() * rhs.norm()));
+}
+
+// Determines whether two maliput::api::LaneEnd::Which are aligned or not.
+// Aligned means that a kStart connects to a kFinish and vice versa.
+bool AreLaneEndsAligned(const maliput::api::LaneEnd::Which& lhs, const maliput::api::LaneEnd::Which& rhs) {
+  return (lhs == maliput::api::LaneEnd::Which::kStart && rhs == maliput::api::LaneEnd::Which::kFinish) ||
+         (lhs == maliput::api::LaneEnd::Which::kFinish && rhs == maliput::api::LaneEnd::Which::kStart);
+}
+
+}  // namespace
+
+// Filters out lanelets that are not considered connected because of having a jump in the heading in their limits.
+// @param lanelet The lanelet to filter out the connected lanelets.
+// @param lane_end_which Lane end of the lanelet where the connections are made.
+// @param connections The connections obtained from the geometrical definition of the map.
+// @param lanelet_layer The lanelet layer to get the connected lanelets.
+// @returns The filtered connections.
+std::unordered_map<Lane::Id, maliput::api::LaneEnd::Which> FilterOutByDirection(
+    const lanelet::Lanelet& lanelet, const maliput::api::LaneEnd::Which& lane_end_which,
+    const std::unordered_map<Lane::Id, maliput::api::LaneEnd::Which>& connections,
+    const lanelet::LaneletLayer& lanelet_layer) {
+  using maliput::api::LaneEnd;
+  const maliput::math::Vector2 tangent_at_end = Get2DTangentAtLaneEnd(lanelet, lane_end_which);
+  std::unordered_map<Lane::Id, maliput::api::LaneEnd::Which> filtered_connections;
+  for (const auto& [connected_lanelet_id, connected_end_which] : connections) {
+    const auto connected_lanelet = lanelet_layer.find(lanelet::Id(std::stoul(connected_lanelet_id)));
```

Comment:
Can you create a lambda to perform the conversion operation?

```C++
auto to_lanelet_id = [](const std::string& connected_lanelet_id) { return lanenlet::Id{std::stoul(connected_lanelet_id); };
...
const auto connected_lanelet = lanelet_layer.find(to_lanelet_id(connected_lanelet_id));
```

Reply:
Just to increase readability.

Reply:
Done.